libs/api/database/src/customTypes/mongoose-fuzzy-searching/index.d.ts

```diff
@@ -0,0 +1,68 @@
+declare module 'mongoose-fuzzy-searching' {
+    import { Document, DocumentQuery, HookAsyncCallback, HookSyncCallback, Model, Schema } from 'mongoose'
+
+    export type FuzzyFieldStringOptions<T> = (keyof T & string)[];
```

Comment:
This basically means "the only fields allowed on this type are the fields already defined on `T`".